src/overlays/actors/ovl_En_Knight/z_en_knight.c

```diff
@@ -205,237 +380,4044 @@ ActorInit En_Knight_InitVars = {
+
+typedef enum {
+    /* 0 */ KNIGHT_SUB_ACTION_RETREAT_0,
+    /* 1 */ KNIGHT_SUB_ACTION_RETREAT_1,
+    /* 2 */ KNIGHT_SUB_ACTION_RETREAT_2,
+    /* 3 */ KNIGHT_SUB_ACTION_RETREAT_3
+} EnKnightSubActionRetreat;
+
+typedef enum {
+    /* 0 */ KNIGHT_SUB_ACTION_RECOIL_0,
+    /* 1 */ KNIGHT_SUB_ACTION_RECOIL_1
+} EnKnightSubActionRecoil;
+
+typedef enum {
+    /* 0 */ KNIGHT_SUB_ACTION_JUMP_BACKWARDS_0,
+    /* 1 */ KNIGHT_SUB_ACTION_JUMP_BACKWARDS_1
+} EnKnightSubActionJumpBackwards;
+
+typedef enum {
+    /* 0 */ KNIGHT_SUB_ACTION_MARCH_0,
+    /* 1 */ KNIGHT_SUB_ACTION_MARCH_1,
+    /* 2 */ KNIGHT_SUB_ACTION_MARCH_2
+} EnKnightSubActionMarch;
+
+typedef enum {
+    /*  0 */ KNIGHT_SUB_ACTION_IGOS_SITTING_0,
+    /*  1 */ KNIGHT_SUB_ACTION_IGOS_SITTING_1,
+    /*  2 */ KNIGHT_SUB_ACTION_IGOS_SITTING_2,
+    /*  5 */ KNIGHT_SUB_ACTION_IGOS_SITTING_5 = 5,
+    /*  6 */ KNIGHT_SUB_ACTION_IGOS_SITTING_6,
+    /* 10 */ KNIGHT_SUB_ACTION_IGOS_SITTING_10 = 10,
+    /* 11 */ KNIGHT_SUB_ACTION_IGOS_SITTING_11
+} EnKnightSubActionIgosSitting;
+
+typedef enum {
+    /* 0 */ KNIGHT_SUB_ACTION_FLYING_HEAD_DONE_0,
+    /* 1 */ KNIGHT_SUB_ACTION_FLYING_HEAD_DONE_1,
+    /* 2 */ KNIGHT_SUB_ACTION_FLYING_HEAD_DONE_2
+} EnKnightSubActionFlyingHeadDone;
+
+typedef enum {
+    /* 0 */ KNIGHT_SUB_ACTION_BREATH_ATTACK_0,
+    /* 1 */ KNIGHT_SUB_ACTION_BREATH_ATTACK_1,
+    /* 2 */ KNIGHT_SUB_ACTION_BREATH_ATTACK_2
+} EnKnightSubActionBreathAttack;
+
+typedef enum {
+    /* 0 */ KNIGHT_SUB_ACTION_FLYING_HEAD_0,
+    /* 1 */ KNIGHT_SUB_ACTION_FLYING_HEAD_1,
+    /* 2 */ KNIGHT_SUB_ACTION_FLYING_HEAD_2,
+    /* 3 */ KNIGHT_SUB_ACTION_FLYING_HEAD_3
+} EnKnightSubActionFlyingHead;
+
+typedef enum {
+    /* 0 */ KNIGHT_SUB_ACTION_FLYING_HEAD_ATTACK_0,
+    /* 1 */ KNIGHT_SUB_ACTION_FLYING_HEAD_ATTACK_1,
+    /* 2 */ KNIGHT_SUB_ACTION_FLYING_HEAD_ATTACK_2
+} EnKnightSubActionFlyingHeadAttack;
+
+typedef enum {
+    /* 0 */ KNIGHT_SUB_ACTION_CAPTAINS_HAT_CS_IGOS_SITTING,
+    /* 1 */ KNIGHT_SUB_ACTION_CAPTAINS_HAT_CS_IGOS_STANDING
+} EnKnightSubActionCaptainsHatCS;
+
+void EnKnight_EffectAdd(PlayState* play, Vec3f* pos, Vec3f* velocity, Vec3f* acceleration, f32 scale, f32 scaleTarget,
+                        s16 alpha) {
+    s16 i;
+    EnKnightEffect* eff;
+
+    for (eff = play->specialEffects, i = 0; i < ARRAY_COUNT(sEnKnightEffects); i++, eff++) {
+        if (!eff->active) {
+            eff->active = true;
+            eff->pos = *pos;
+            eff->velocity = *velocity;
+            eff->acceleration = *acceleration;
+            eff->unk_30 = 0;
+            eff->scale = (BREG(50) * 0.0001f + 0.001f) * scale;
+            eff->scaleTarget = (BREG(51) * 0.0001f + 0.001f) * scaleTarget;
+            eff->alpha = alpha;
+            eff->scrollTimer = 0;
+            break;
+        }
+    }
+}
+
+void EnKnight_UpdateLimbCollider(EnKnight* this, s32 elemNum, ColliderJntSph* jntSph, Vec3f* pos) {
+    f32 scale = (this == sIgosInstance) ? 1.3076924f : 1.0f;
```

Comment:
Does the `(17.0f / 13)` you used for the blure function not work here?

Reply:
It doesn't unfortunately

Reply:
What the heck, lol. Oh well, thanks for checking.